spec/helpers/time_options_helper_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
+
+RSpec.describe TimeOptionsHelper do
+  describe "#time_options" do
+    subject { helper.time_options }
+
+    it "contains a 12:01am special case" do
+      expect(subject).to include "12:01am"
+    end
+
+    it "does not contain 12:00am special case" do
+      expect(subject).not_to include "12:00am"
+    end
+
+    it "returns 48 items (hourly - except 12am special case - and half-hourly)" do
+      expect(subject.count).to eq(48)
+    end
+
+    it "returns all the AM times before the PM times" do
+      index_of_last_am = subject.each_with_index.map { |time, index| time.end_with?("am") ? index : nil  }.compact.last
+      index_of_first_pm = subject.index { |time| time.end_with?("pm") }
+      expect(index_of_last_am).to be < index_of_first_pm
+    end
+
+    it "should only contain time values" do
+      subject.each do |time|
+        expect(time).to match /[0-9]{1,2}:[0-9]{2}(am|pm)/
```

Comment:
Regular Expressions are difficult to read. This is one time where a comment explaining the expected outcomes would be useful.

Reply:
Have now changed to multiline regex with comments for each part of the regex 👍